catalog/dags/providers/provider_api_scripts/smk.py

```diff
@@ -87,41 +86,39 @@ def _get_creator(item: dict) -> str | None:
 
     @staticmethod
     def _get_images(item: dict) -> list:
-        images = []
-
         # Legacy images do not have an iiif_id; fall back to the ID from the
         # collection DB.
         iiif_id = item.get("image_iiif_id")
-        image_id = iiif_id or item.get("id")
-
-        if image_id is not None:
-            if iiif_id is None:
-                # Legacy images do not have IIIF links.
-                image_url = item.get("image_native")
-            else:
-                image_url = SmkDataIngester._get_image_url(iiif_id)
-
-            thumbnail_url = item.get("image_thumbnail")
-            height = item.get("image_height")
-            width = item.get("image_width")
-            filesize = item.get("image_size") or item.get("size")
-            images.append(
-                {
-                    "id": image_id,
-                    "image_url": image_url,
-                    "thumbnail_url": thumbnail_url,
-                    "height": height,
-                    "width": width,
-                    "filesize": filesize,
-                }
-            )
+
+        if not (foreign_identifier := iiif_id or item.get("id")):
+            return []
+
+        if iiif_id:
+            image_url = SmkDataIngester._get_image_url(iiif_id)
+        else:
+            # Legacy images do not have IIIF links.
+            image_url = item.get("image_native")
+
+        thumbnail_url = item.get("image_thumbnail")
+        height = item.get("image_height")
+        width = item.get("image_width")
+        filesize = item.get("image_size") or item.get("size")
 
         # We used to get additional images from the `alternative_images` field,
         # but we found these to be either redundant, duplicates, or lower quality
         # versions. See https://github.com/WordPress/openverse-catalog/issues/875
         # for the full investigation & discussion
```

Comment:
The method signature here confused me at first, but this comment explains it; this used to build a list including related images. Now that we only ever return one image, I think we should get rid of this method and pull this logic into `get_record_data`.

This isn't related to this refactor so we can do that in another PR, but it stood out to me so worth mentioning!